Document new logic of `universalReceiver(..)` function in LSP0/9 (#134)
lukso-network/LIPs: LSPs/LSP-0-ERC725Account.md

- [LSP1](./LSP-1-UniversalReceiver.md#specification)
- [LSP14](./LSP-14-Ownable2Step.md#specification)

#### universalReceiver

```solidity
function universalReceiver(bytes32 typeId, bytes memory data) public payable returns (bytes memory)
```

This function is part of the [LSP1-UniversalReceiver] Specification, and do the following:

- Emits [ValueReceived](#events) event when receiving native tokens.

> - Forwards the call to the `universalReceiverDelegate(..)` function in the **UniversalReceiverDelegate** contract which address is stored under the data key attached below, if it supports [LSP1UniversalReceiverDelegate InterfaceId](../smart-contracts/interface-ids.md). If there is no address stored under the data key, execution continue normally.

Reviewer: _Suggested rewording_

```suggestion
- if an `address` is stored under the data key [`LSP1UniversalReceiverDelegate`](./LSP-1-UniversalReceiver.md#universalreceiverdelegate) and this address is a contract that supports the [LSP1UniversalReceiverDelegate InterfaceId](../smart-contracts/interface-ids.md), forwards the call to the `universalReceiverDelegate(..)` function of the **UniversalReceiverDelegate** contract. If there is no address stored under this data key, execution continues normally.
```

I would suggest to delete the schema below and just add a link to the LSP1 standard page, as it will make it easier to maintain:

```
> For more details about the ERC725Y JSON Schema of the `LSP1UniversalReceiverDelegate` data key, see standard page [**LSP 1 - Universal Receiver > UniversalReceiverDelegate**](./LSP-1-UniversalReceiver.md#universalreceiverdelegate)
```

```json
{
  "name": "LSP1UniversalReceiverDelegate",
  "key": "0x0cfc51aec37c55a4d0b1a65c6255c4bf2fbdf6277f3cc0730c45b828b6db8b47",
  "keyType": "Singleton",
  "valueType": "address",
  "valueContent": "Address"
}
```

- Forwards the call to the `universalReceiverDelegate(..)` function in the **TypeIdDelegate** contract which address is stored under the data key attached below, if it supports [LSP1UniversalReceiverDelegate InterfaceId](../smart-contracts/interface-ids.md). If there is no address stored under the data key, execution continue normally.

```json

> {

Reviewer: This schema however we need to keep it here, as I guess the typeId routing is specific to LSP0 + LSP9.

Or should this type ID be part of the LSP1 standard? (like an other type of extension for LSP1).

I think it's better to keep it for LSP0/9 only, as it is only a specific behaviour we want only on these standard, but could be discussed if anyone has other arguments. @YamenMerhi @frozeman @b00ste

  "name": "LSP1UniversalReceiverDelegate:<bytes32>",

>   "key": "0x0cfc51aec37c55a4d0b10000<bytes32>",

Reviewer 2: Wouldn’t appending a `bytes32` value after `0x0cfc51aec37c55a4d0b10000` make the key invalid? Shouldn’t we append the first 20 bytes of the type id , meaning `bytes20`?
